service/history/execution/mutable_state_builder.go

```diff
@@ -2369,28 +2336,53 @@ func (e *mutableStateBuilder) logDuplicatedActivityEvents(events []*types.Histor
 				startedEventID:   event.ActivityTaskTimedOutEventAttributes.StartedEventID,
 			}
 		default:
-			return
+			return true
 		}
 
 		if _, ok := activityTaskUniqueEvents[uniqueEventParams]; ok {
-			e.logger.Error("Duplicate activity task event found",
+			e.logger.Warn("Duplicate activity task event found",
```

Comment:
nit: I would still keep it as Error level since we did performed some risky change on this. For all our clusters, we only have 12 such occurrences in past 7 days. So this wouldn't be catastrophic of keep it error level